aws_lambda_powertools/utilities/parser/envelopes/kinesis.py

```diff
@@ -38,5 +38,6 @@ def parse(self, data: Optional[Union[Dict[str, Any], Any]], model: Type[Model])
         parsed_envelope: KinesisDataStreamModel = KinesisDataStreamModel.parse_obj(data)
         logger.debug(f"Parsing Kinesis records in `body` with {model}")
         return [
-            self._parse(data=record.kinesis.data.decode("utf-8"), model=model) for record in parsed_envelope.Records
+            self._parse(data=cast(bytes, record.kinesis.data).decode("utf-8"), model=model)
```

Comment:
can we solve without casting?

Reply:
Yes, I can explicitly declare local variable and check if record.kinesis.data is byte type and only then do this translation - otherwise do something else. What should we do in case that user pass another data  model in data key as allowed here: 
https://github.com/awslabs/aws-lambda-powertools-python/blob/0196883410bf4fadeb1ca27f3f331ec6fbaffbe4/aws_lambda_powertools/utilities/parser/models/kinesis.py#L17

Reply:
Refactor code to use temp variable with comment:

```
        for record in parsed_envelope.Records:
            # We allow either AWS expected contract (str) or a custom Model, see #943
            body = cast(str, record.body)
            sns_notification = SnsNotificationModel.parse_raw(body)
            output.append(self._parse(data=sns_notification.Message, model=model))
```